yb-voyager/src/query/queryissue/detectors.go

```diff
@@ -555,6 +556,63 @@ func (j *JsonPredicateExprDetector) GetIssues() []QueryIssue {
 	return issues
 }
 
+type NonDecimalIntegerLiteralDetector struct {
+	query    string
+	detected bool
+}
+
+func NewNonDecimalIntegerLiteralDetector(query string) *NonDecimalIntegerLiteralDetector {
+	return &NonDecimalIntegerLiteralDetector{
+		query: query,
+	}
+}
+func (n *NonDecimalIntegerLiteralDetector) Detect(msg protoreflect.Message) error {
+	if queryparser.GetMsgFullName(msg) != queryparser.PG_QUERY_ACONST_NODE {
+		return nil
+	}
+	aConstNode, err := queryparser.ProtoAsAConstNode(msg)
+	if err != nil {
+		return err
+	}
+	/*
+		Caveats can't report this issue for cases like -
```

Comment:
Let's also add a comment explaining where this is useful then - i guess only plpgsql objects?

Reply:
yeah majorly plpgsql in normal cases, will add a comment